### ISSUE TITLE
Add comments

### ISSUE THREAD

**Issue by fathomssen** [fathomssen/redtimer#24]
Add a text field to write comments which will be appended to the issue.

**Comment by Warlib1975** [comment]
It's really quite important thing when a developer or a tester ad some comment when they finish do the job.

**Comment by TyRoden** [comment]
I have to agree.. time entries with no comments make this nearly unusable for us.. this is a great tool and would like to use it.

**Comment by fathomssen** [comment]
Where would you suggest the comment field should be? Right beneath the timer, above the activity selection?

**Comment by TyRoden** [comment]
I would say below the timer.. thats where I naturally would look for it.

**Comment by fathomssen** [comment]
Sounds reasonable. What about the case when you (like I) use keyboard shortcuts most of the time?

Should the last value of the comment field be used (does that even make sense?) or should the comment field be left empty?

Or should you get a pop-up window asking you for a comment? This might not work properly in environments with multiple desktops (you simply would not notice the new window).

**Comment by Warlib1975** [comment]
I think comment field should be empty. The process are: 
1. Developer turn on the timer. 
2. When he has finished the job, he writes comments and stop timer. 
3. It would be good if it was possibility to stop timer, then write comment and save it. Just for any case, if developer forgot the sequence. 

Comment window may be instead of issue description. E.g. when timer is started description tab switched to comment tab.

I believe that there is no any additional pop-up window, since comment - it's not obliging field.

**Comment by fathomssen** [comment]
There is no easy way to amend a time booking after it has been sent to Redmine. So it will be required to enter the comment before stopping the timer.

I would suggest the following implementation:

1. There will be a new configuration option in the settings dialog that specifies whether a comment is mandatory.
2. If the comment is NOT mandatory, it MAY be typed into the comment fields and will be saved. If it is left empty, that's OK as well.
3. If the comment IS mandatory and it has not been specified in the comment field, a message will pop up upon stopping the timer. The timer will continue to count until a comment has been specified and the timer has been stopped again.

What do you think about this approach?

Best regards
Frederick Thomssen

**Comment by Warlib1975** [comment]
I believe it's a good decision. :-)

**Comment by TyRoden** [comment]
I like the switch for comment mandatory or not. If mandatory, merely popup
the app from the tray and show the comment field with the cursor sitting
ready (that way if using shortcuts it works fine)

**Comment by kristian-b** [comment]
Adding a comment is very important for me too. Would be great if that would be implemented.
